### PR TITLE
rename files and add readme for llama auto_parallel

### PR DESCRIPTION
**Pull request by zhiqiu**
### PR types

Others
### PR changes

Docs
### Description

rename files and add readme for llama auto_parallel

### CODE REVIEW

File: paddlenlp/transformers/llama/modeling_auto.py [hunk 1]
```diff
@@ -1222,20 +1221,21 @@ def forward(
...
-            return (loss,) + output if loss is not None else output
-
-        return CausalLMOutputWithCrossAttentions(
-            loss=loss,
-            logits=logits,
-            past_key_values=outputs.past_key_values,
-            hidden_states=outputs.hidden_states,
-            attentions=outputs.attentions,
-        )
+        # loss = None
```
Comment (ZHUI) [inline paddlenlp/transformers/llama/modeling_auto.py:1226]: 注释的直接删除？

File: paddlenlp/transformers/llama/modeling_auto.py [hunk 2]
```diff
@@ -744,6 +727,42 @@ def get_tensor_parallel_split_mappings(num_layers):
...
+        # Layer.apply is DFS https://github.com/PaddlePaddle/Paddle/blob/a6f5021fcc58b21f4414bae6bf4731ef6971582c/python/paddle/nn/layer/layers.py#L527-L530
+        # sublayer is init first
+        # scale RowParallelLinear weight
+        with paddle.no_grad():
+            if isinstance(layer, LlamaMLPAuto):
+                factor = 1 / math.sqrt(2 * self.config.num_hidden_layers)
+                layer.down_proj.weight.scale_(factor)
+            if isinstance(layer, LlamaAttentionAuto):
+                factor = 1 / math.sqrt(2 * self.config.num_hidden_layers)
+                layer.o_proj.weight.scale_(factor)
```
Comment (ZHUI) [inline paddlenlp/transformers/llama/modeling_auto.py:763]: 删除？

File: paddlenlp/transformers/llama/modeling_auto_static.py [hunk 3]
```diff
@@ -1041,29 +1042,19 @@ def forward(self, prediction_scores, masked_lm_labels):
...
-
-            masked_lm_loss = self.loss_func(
-                prediction_scores.astype("float32"),
-                masked_lm_labels.unsqueeze(2),
-            )
+        # with paddle.amp.auto_cast(False):
+        #     masked_lm_loss = self.loss_func(prediction_scores.astype("float32"), masked_lm_labels.unsqueeze(2))
+        #     # skip ignore_index which loss == 0
+        #     masked_lm_loss = masked_lm_loss[masked_lm_loss > 0].astype("float32")
+        #     loss = paddle.mean(masked_lm_loss)
```
Comment (ZHUI) [inline paddlenlp/transformers/llama/modeling_auto_static.py:1049]: 删除？

**Review by ZHUI — APPROVED** [review]
LGTM

File: paddlenlp/transformers/llama/modeling_auto.py (hunk 1 above)
Comment (zhiqiu) [reply paddlenlp/transformers/llama/modeling_auto.py:1226]: 后面提PR完善代码

**Pull request merged**